docs/dev/design/new-notifications-system.rst

```diff
@@ -0,0 +1,459 @@
+            body=_(
+                """
+          There was a problem with Read the Docs while building your documentation.
+          Please try again later.
+          If this problem persists,
+          report this error to us with your build id ({instance[pk]}).
+        """,
+                type=ERROR,
+            ),
+        ),
+        "build-os-required": Message(
+            header=_("Invalid configuration"),
+            body=_(
+                """
+          The configuration key "build.os" is required to build your documentation.
+          <a href='https://docs.readthedocs.io/en/stable/config-file/v2.html#build-os'>Read more.</a>
+        """,
+                type=ERROR,
+            ),
+        ),
+        "cancelled-by-user": Message(
+            header=_("User action"),
+            body=_(
+                """
+          Build cancelled by the user.
+        """,
+                type=ERROR,
+            ),
+        ),
+        "os-ubuntu-18.04-deprecated": Message(
+            header=_("Deprecated OS selected"),
+            body=_(
+                """
+          Ubuntu 18.04 is deprecated and will be removed soon.
+          Update your <code>.readthedocs.yaml</code> to use a newer image.
+        """,
+                type=TIP,
+            ),
+        ),
+    }
+
+
+
+``Notification`` model definition
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+This class is the representation of a notification attached to an resource (e.g. User, Build, etc) in the database.
+It contains an identifier (``slug``) pointing to one of the messages defined in the previous section (key in constant ``NOTIFICATION_MESSAGES``).
+
+.. code-block:: python
+
+    import textwrap
+    from django.utils.translation import gettext_noop as _
+
+
+    class Notification(TimeStampedModel):
+        # Message identifier
+        slug = models.CharField(max_length=128)
+
+        # UNREAD: the notification was not shown to the user
+        # READ: the notifiation was shown
+        # DISMISSED: the notification was shown and the user dismissed it
+        state = models.CharField(
+            choices=[UNREAD, READ, DISMISSED],
+            default=UNREAD,
+            db_index=True,
+        )
+
+        # Makes the notification imposible to dismiss (useful for Build notifications)
+        dismissable = models.BooleanField(default=False)
+
+        # Show the notification under the bell icon for the user
+        bell = models.BooleanField(default=False, help_text="Show under bell icon")
+
+        # Notification attached to
+        #
+        # Uses ContentType for this.
+        # https://docs.djangoproject.com/en/4.2/ref/contrib/contenttypes/#generic-relations
+        #
+        attached_to_content_type = models.ForeignKey(ContentType, on_delete=models.CASCADE)
+        attached_to_id = models.PositiveIntegerField()
+        attached_to = GenericForeignKey("attached_to_content_type", "attached_to_id")
```

Comment:
This would be good to get more feedback on. I have used generic relations in the past, but perhaps there is a strong reason not to.

Reply:
I've used it in the past as well. I will give it a try and decide what to do at the implementation time depending on how comfortable it feels 😄